mlir/lib/Catalyst/Transforms/DetectQNodes.cpp

```diff
@@ -733,7 +733,7 @@ void replaceTerminatorWithUnconditionalJumpToSuccessBlock(SmallVector<Block *> a
         PatternRewriter::InsertionGuard insertGuard(rewriter);
         auto terminator = abort->getTerminator();
         rewriter.setInsertionPoint(terminator);
-        assert(isa<LLVM::UnreachableOp>(terminator));
+        assert(isa<LLVM::UnreachableOp>(terminator) || isa<LLVM::BrOp>(terminator));
```

Comment:
Got it. Maybe we can just remove the assertion but happy to leave it as well :) Thanks for fixing this!